components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -119,15 +114,32 @@ impl CanvasContextState {
     }
 }
 
+#[must_root]
 #[derive(JSTraceable, MallocSizeOf)]
 pub struct CanvasState {
     #[ignore_malloc_size_of = "Defined in ipc-channel"]
     ipc_renderer: IpcSender<CanvasMsg>,
     canvas_id: CanvasId,
+    state: DomRefCell<CanvasContextState>,
+    origin_clean: Cell<bool>,
+    canvas: Option<&HTMLCanvasElement>,
+    #[ignore_malloc_size_of = "Arc"]
+    image_cache: Arc<dyn ImageCache>,
+    base_url: ServoUrl,
+    global: &GlobalScope,
```

Comment:
Remove `global` and have callers pass in a `&GlobalScope` argument when necessary.